pkg/cloud/services/networking/port.go

```diff
@@ -243,6 +244,32 @@ func (s *Service) DeletePort(eventObject runtime.Object, portID string) error {
 	return nil
 }
 
+func (s *Service) DeletePorts(openStackCluster *infrav1.OpenStackCluster) error {
+	networkID := openStackCluster.Spec.Network.ID
+	portList, err := s.client.ListPort(ports.ListOpts{
+		NetworkID:   networkID,
+		DeviceOwner: "",
+	})
+	if err != nil {
+		if capoerrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("list ports of network %q: %v", networkID, err)
+	}
+
+	for _, port := range portList {
+		if strings.Contains(port.Name, openStackCluster.Name) {
```

Comment:
```suggestion
		if strings.HasPrefix(port.Name, openStackCluster.Name) {
```

`HasPrefx` will be enough as the port name ist prefixed by the instance name (and the instance name is prefixed by the cluster name :))

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/7a8c2e82b58c66ca9a20bbf22be916c956b51998/pkg/cloud/services/compute/instance.go#L279-L285

Reply:
yes, thanks, I have changed it,please have a re-check